### ISSUE TITLE
ICLR '23 | Aligning Model and Macaque Inferior Temporal Cortex Representations Improves Model-to-Human Behavioral Alignment and Adversarial Robustness.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#73]
Dapello et. al. [Aligning Model and Macaque Inferior Temporal Cortex Representations Improves Model-to-Human Behavioral Alignment and Adversarial Robustness](https://openreview.net/forum?id=SMYdcXjJh1q).

**Comment by NorbertZheng** [comment]
## Overview
While
 - some state-of-the-art artificial neural network systems in computer vision are strikingly accurate models of the corresponding primate visual processing,

there are still many discrepancies between these models and the behavior of primates on **object recognition tasks*.

 - Many current models suffer from extreme sensitivity to adversarial attacks and often do not align well with the image-by-image behavioral error patterns observed in humans.

Previous research has provided strong evidence that primate object recognition behavior can be very accurately predicted by neural population activity in the **inferior temporal (IT) cortex**, a brain area in the late stages of the visual processing hierarchy. Therefore, here we directly test
 - whether making the late-stage representations of models more similar to that of macaque IT produces new models that exhibit more robust, primate-like behavior.

We conducted chronic, large-scale multi-electrode recordings across the IT cortex in six non-human primates (rhesus macaques). We then use these data to fine-tune (end-to-end) the model "IT" representations such that
 - they are more aligned with the biological IT representations while preserving accuracy on object recognition tasks.

We generate a cohort of models with a range of IT similarity scores validated on held-out animals across two image sets with distinct statistics.
 - Across a battery of optimization conditions, we observed a strong correlation between the models' IT likeness and alignment with human behavior, as well as an increase in its adversarial robustness.

We further assessed the limitations of this approach and find that 
 - the improvements in behavioral alignment and adversarial robustness generalize across different image statistics, but not to ~~object categories outside of those covered in our IT training set~~.

Taken together, our results demonstrate that building models that are more aligned with the primate brain leads to more robust and human-like behavior, and call for larger neural data sets to further augment these gains.

**Comment by NorbertZheng** [comment]
## Model
![image](https://user-images.githubusercontent.com/37235023/223307104-c171215f-e3e6-4859-a0b7-9540dfe2b874.png)
Aligning model IT representations with primate IT representations improves behavioral alignment and improves adversarial robustness.
 - A) A set of naturalistic images, each containing one of eight different object classes are shown to a CNN and also to three different primate subjects with implanted multi-electrode arrays recording from the Inferior Temporal (IT) cortex. 
    - (1) A Base model (ImageNet pre-trained CORnet-S) is fine-tuned using stochastic gradient descent to
    - (2) minimize the classification loss with respect to the ground truth object in each image while also minimizing a representational similarity loss (CKA) that encourages the model’s IT representation to be more like those measured in the (pooled) primate subjects.
    - (3) The resultant IT aligned models are then frozen and each tested in three ways.
       - First, model IT representations are evaluated for similarity to biological IT representation (CKA metric) using neural data obtained from new primate subjects -- we refer to the split-trial reliability ceiled average across all held out macaques and both image sets as "Validated IT neural similarity". 
       -  Second, model output behavioral error patterns are assessed for alignment with human behavioral error patterns at the resolution of individual images (i2n, see Methods).
       - Third, model behavioral output is evaluated for its robustness to white box adversarial attacks using an $L_{\infty}$ norm projected gradient descent attack.

      All three tests are carried out with: (i) new images within the IT-alignment training domain (held out HVM images; see Methods) and (ii) new images with novel image statistics (natural COCO images; see Methods), and those empirical results are tracked separately.
 - B) We find that this IT-alignment procedure produced gains in validated IT neural similarity relative to base models on both data sets, and that these gains led to improvement in human behavioral alignment. n=30 models are shown, resulting from training at six different relative weightings of the IT neural similarity loss, each from five base models that derived from five random seeds.
 - C) We also find that these same IT-alignment gains resulted in increased adversarial accuracy (PGD $L_{\infty}$, $\epsilon=\frac{1}{1020}$) on the same model set as in B. Base models trained only for ImageNet and HVM image classification are circled in grey.

**Comment by NorbertZheng** [comment]
## Results
### IT Representational Similarity
![image](https://user-images.githubusercontent.com/37235023/223308165-731d8980-2713-49a0-afec-01ec8049003b.png)
IT alignment training leads to improved IT representational similarity on held out animals and held out images across two image sets with different statistics.
 - A) IT neural similarity scores (CKA, normalized by split-half trial reliability) for held out but within domain HVM images vs gradient steps is shown for two held out monkeys across seven different neural similarity loss gradient dropout rates (the darkest trace receives neural similarity loss gradients at 100% of gradient steps, while in the lightest trace neural similarity loss gradients are dropped at every step). Two control conditions are also shown: optimizing model IT toward a random Gaussian target IT matrix (random, blue) and toward an image-shuffled target IT matrix (shuffle, orange).
 - B) Like A but for natural COCO images out of domain with respect to the training set. Grey dashed line on each plot shows the base model score for models pre-trained on ImageNet and HVM image labels with no IT representational similarity loss, which the model set with 0% of IT similarity loss gradients does not deviate significantly from. Error bars are bootstrapped confidence intervals for 5 training seeds.

**Comment by NorbertZheng** [comment]
### Behavioral Alignment
![image](https://user-images.githubusercontent.com/37235023/223308510-09d7e454-0655-48cf-86eb-addd940c7c9b.png)
IT neural similarity correlates with behavioral alignment across a variety of optimization conditions and unseen image statistics but not on unseen object categories.
 - A) Held out animal and image IT neural similarity is plotted against human behavioral alignment on the HVM image set at every validation epoch for all neural loss weight conditions, random Gaussian IT target matrix conditions, and image shuffled IT target matrix conditions, in each case with or with and with out image classification loss.
 - B) and C)  Like in A but for the COCO image set and the Objectome image set Rajalingham et al. (2018) filtered to overlapping categories with the IT training set.
 - D) The behavioral alignment for the full Objectome image set with 20 categories not covered in the IT training is not improved by the IT-alignment procedure and data used here. In all plots, the black cross represents the average base model position, and the heavy blue line is a sliding X, Y average of all conditions merely to visually highlight trends. Five seeds for each condition are plotted.

**Comment by NorbertZheng** [comment]
### White Box Adversarial Robustness
![image](https://user-images.githubusercontent.com/37235023/223308869-ca033ece-33ae-48dd-8383-64bd94943fc0.png)
IT neural similarity correlates with improved white box adversarial robustness.
 - A) held out animal and image IT neural similarity is plotted against white box adversarial accuracy (PGD $L_{\infty}$ $\epsilon=\frac{1}{1020}$) on the HVM image set measured across multiple training time points for all neural loss ratio conditions, random Gaussian IT target matrix conditions, and image shuffled IT target matrix conditions.
 - B) Like in A but for COCO images. In both plots, the black cross represents the average base model position, the black X marks a CORnet-S adversarially trained on HVM images, and the heavy blue line is a sliding X, Y average of all conditions merely to visually highlight trends. Five seeds for each condition are plotted.

![image](https://user-images.githubusercontent.com/37235023/223309295-8b711525-22c0-493a-bf26-b5ac1b580313.png)
IT aligned models are more robust than standard models in and out of domain, and more robust than adversarially trained models in out of domain conditions.
 - A) PGD $L_{\infty}$ and $L_{2}$ strength accuracy curves on HVM images for standard trained networks (green) IT aligned networks (blue) and networks adversarially trained (PGD $L_{\infty}$ $\epsilon=\frac{4}{255}$) on the IT fitting image labels (orange).
 - B) Like in A but for COCO images. Error shading represents bootstrapped 95% confidence intervals over five training seeds.

**Comment by NorbertZheng** [comment]
## Conclusion
 - Our method robustly improves IT representational similarity of models to brains even when measured on new animals and new images.
 - We find that gains in model IT-likeness lead to gains in human behavioral alignment.
 - Likewise we find that improved IT-likeness leads to increased adversarial robustness.
